debezium-core/src/main/java/io/debezium/util/UlidIdBuilder.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.util;
+
+import de.huxhorn.sulky.ulid.ULID;
+
+import java.util.Optional;
+
+public class UlidIdBuilder extends OrderedIdBuilder {
+    private final ULID idGen = new ULID();
+    private Clock clock = Clock.SYSTEM;
+
+    protected ULID.Value lastId = new ULID.Value(0, 0);
+
+    @Override
+    public Boolean shouldIncludeId() {
+        return true;
+    }
+
+    @Override
+    public String buildNextId() {
+        Optional<ULID.Value> n = next();
+        while(!n.isPresent()) {
+            // we just try and sleep here to next ms where
+            // we likely won't get a problem
+            // and ignore any interrupts, as we can just try
+            // again on the next loop
+            try {
+                Thread.sleep(1);
```

Comment:
Ah, I think I misunderstood this part before. It will only sleep in case there _is_ a collision already, right?

Reply:
correct

Reply:
Also, just as a reminder that should be _exceedingly_ rare, like it may never happen in reality, but we handle it anyways